reporters/plain_reporter.py

```diff
@@ -12,8 +13,52 @@ def handle_message(self, msg):
 
     def print_message_ids(self):
         # Sort the messages by their type.
-        self._messages.sort(key=lambda s: s[0])
-
+        #self._messages.sort(key=lambda s: s[0])
+        self.sort_messages()
         for msg in self._messages:
             code = msg.msg_id
             print(code, '({})\n    {}'.format(msg.symbol, msg.msg))
+
+    def sort_messages(self):
+        # Sort the messages by their type.
+        self._messages.sort(key=lambda s: s[0])
+
+        i = 0
+        while i < len(self._messages):
+            current_id = self._messages[i].msg_id
+            count = 1
+            messages = ''
+            msg_count = 1
+            while i+1 < len(self._messages) and self._messages[i+1].msg_id == current_id:
+                count += 1
+                if msg_count <= 5:
+                    messages += '\n    ' + self._messages[i+1].msg
+                    msg_count += 1
+                self._messages.pop(i+1)
+
+            msg = self._messages[i].msg
+
+            if count > 1:
+                msg = 'This error occurs at '+ str(count) + ' places:\n    ' + self._messages[i].msg + messages
```

Comment:
As we discussed in the meeting, this string should not appear here as a message, but only when the messages are reported.

It is likely necessary to create a new attribute, or use an existing one, to store the counts for each message type so that they can be reported.